docker/index.js

```diff
@@ -2,100 +2,241 @@ const express = require('express');
                input path: /apps/sf_ainft/sd_inpainting/$user_addr/$tokenId/$timestamp/input
                signed data path: /apps/sf_ainft/sd_inpainting/$user_addr/$tokenId/$timestamp/signed_data
                `)
-        .end();
+		.end();
 });
 
 app.post('/trigger', async (req, res) => {
 
-    // Example of the transaction shape: refer to tx_sample.json
-
-    // 1. check tx meets precondition
-    const tx = req.body.transaction;
-    if (!tx || !tx.tx_body || !tx.tx_body.operation) {
-        console.log(`Invalid tx: ${JSON.stringify(tx)}`);
-        return;
-    }
-    if (tx.tx_body.operation.type !== 'SET_VALUE') {
-        console.log(`Not supported tx type: ${tx.tx_body.operation.type}`)
-        return;
-    }
-
-    const inputPath = tx.tx_body.operation.ref;
-    const parsedInputPath = parsePath(inputPath);
-    if (parsedInputPath.length !== 7 ||
-        parsedInputPath[0] !== 'apps' ||
-        parsedInputPath[1] !== 'sf_ainft_0' ||
-        parsedInputPath[2] !== 'sd_inpainting' ||
-        parsedInputPath[6] !== 'input') {
-        console.log(`Not supported path pattern: ${inputPath}`);
-        return;
-    }
-
-    // 2. call GET /tasks/{task_id}
-    const inputValue = tx.tx_body.operation.value;
-    const options = JSON.parse(inputValue);
-
-    const task_id = options.task_id;
-    const pickedOptions = (({ prompt, seed, guidance_scale }) => ({ prompt, seed, guidance_scale }))(options);
-    pickedOptions = {...pickedOptions, ...{"num_images_per_prompt": 1}};
-    
-    // pickedOptions = {
-    //     prompt: ...,
-    //     seed: ...,
-    //     guidance_scale: ...,
-    //     num_images_per_prompt: 1
-    // }
-    
-    const SDResult = await axios.get(`${SD_INPAINTING_ENDPOINT}/tasks/${task_id}`, pickedOptions);
-    console.log(JSON.stringify(SDResult.data,null,2));
-
-    if (SDResult.data.status !== "completed") {
-        console.log(`Task ${task_id} is not completed!`);
-        res.send(`Task ${task_id} is not completed!`);
-        return;
-    }
-    
-
-    //pre-check the output path
-    const outputPath = formatPath([...parsedInputPath.slice(0, parsedInputPath.length - 1), "signed_data"]);
-    const result = await ain.db.ref(outputPath).setValue({
-      value: `${JSON.stringify(SDResult.data, null, 2)}`,
-      nonce: -1,
-    })
-    .catch((e) => {
-      console.error(`setValue failure:`, e);
-    });
-  
-});
+	const { transaction } = req.body;
+	const value = transaction.tx_body.operation.value;
+	const taskId = value.task_id;
+	if(cache.get(taskId)){
+		cache.ttl(taskId, 60);
+		return;
+	}
+
+	// if request is first request, set cache 
+	cache.set(taskId, true, 60);
+	
+
+	let uploadMetadataRes; // for catch error
+
+	const inputPath = transaction.tx_body.operation.ref;
+	const parsedInputPath = parsePath(inputPath);
+
+	// pre-check the output path
+	const outputPath = await formatPath([...parsedInputPath.slice(0, parsedInputPath.length - 1), "signed_data"]);
```

Comment:
How about changing `signed_data` into `verify` as described in the metadata documentation?

Reply:
Hmm.. i think "verify" is simpler than "signed_data". How about @clu3xi ?

Reply:
agree